device/cuda/src/ambiguity_resolution/greedy_ambiguity_resolution_algorithm.cu

```diff
@@ -458,10 +458,55 @@ greedy_ambiguity_resolution_algorithm::operator()(
         nThreads_scan = scan_dim.first;
         nBlocks_scan = scan_dim.second;
 
+        /*
+        CUDA kernel sequence with multiple streams
+        │
+        ├── reset_status
+        │
+        ├── find_max_shared
+        │
+        ├── remove_tracks
+        │       │
+        │       └── [record event_removal]
+        │
+        ├── sort_updated_tracks  (Main stream — executed after event_removal)
+        │       │
+        │       └── [record event_main]
+        │
+        ├───▶ stream_fill
+        │       │
+        │       └── [wait for event_removal]
+        │       │
+        │       └── fill_inverted_ids
+        │               │
+        │               └── [record event_fill]
+        │
+        ├───▶ stream_scan
+        │       │
+        │       └── [wait for event_removal]
+        │       │
+        │       ├── block_inclusive_scan
+        │       ├── scan_block_offsets
+        │       └── add_block_offset
+        │               │
+        │               └── [record event_scan]
+        │
+        ├── [wait for event_main, event_fill, and event_scan] ← sync point
+        │
+        ├── rearrange_tracks
+        │
+        └── gather_tracks
+        */
+
         // Make CUDA Graph
         cudaGraph_t graph;
         cudaGraphExec_t graphExec;
 
+        // Create stream for parallelizable kernels
+        cudaStream_t stream_fill, stream_scan;
+        cudaStreamCreateWithFlags(&stream_fill, cudaStreamNonBlocking);
+        cudaStreamCreateWithFlags(&stream_scan, cudaStreamNonBlocking);
```

Comment:
Not only do you create streams inside of the execute function of the algorithm (as opposed to its constructor), but you do this inside of the while-loop of the algorithm.

Plus I don't see any `cudaStreamDestroy(...)` statements.